retrieval/scrape.go

```diff
@@ -833,12 +877,9 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	re, ok := sl.refCache[s]
-
+	ref, ok := sl.cache.getRef(s)
```

Comment:
I can see it being valid if they had metric_relabel_configs to make it work out. Definitely weird, but I think it's something that should work.